docs/static/monitoring-apis.asciidoc

```diff
@@ -0,0 +1,312 @@
+[[monitoring]]
+== Monitoring APIs
+
+Logstash provides the following monitoring APIs to retrieve runtime metrics
+about Logstash:
+
+* <<hot-threads-api>>
+* <<root-resource-api>>
+* <<stats-info-api>>
+
+//NOTE: Need to add this to the doc after Alpha 1: * <<plugins-api>>
+
+=== Testing Logstash APIs in Sense
+
+//QUESTION: What do you think of this approach? I felt that the links for invoking Sense would be useful, but I wanted to avoid confusing users. If they don't notice this topic, they might still be confused. Do you think I should keep the links that launch Sense, or should I remove them?
+
+Sense is a Kibana app that you can use to build and test REST calls. You'll notice
+that the monitoring API documentation includes links for viewing the examples
+in Sense. For example:
+
+[source,js]
+--------------------------------------------------
+GET /
+--------------------------------------------------
+//AUTOSENSE
+
+After clicking the link to view the example, you need to change the address
+in the Server field to point to port `9600` before you can successfully test the
+monitoring APIs. 
+
+For example, if Kibana is running on localhost, you need to change the value
+of the Server field from `http://localhost:9200` to `http://localhost:9600`.
+
+[[monitoring-common-options]]
+=== Common Options
+
+The following options can be applied to all of the Logstash monitoring APIs.
+
+[float]
+=== Pretty Results
+
+When appending `?pretty=true` to any request made, the JSON returned
+will be pretty formatted (use it for debugging only!). Another option is
+to set `?format=yaml` which will cause the result to be returned in the
+(sometimes) more readable yaml format.
+
+[float]
+=== Human-Readable Output
+
+Statistics are returned in a format suitable for humans
+(eg `"exists_time": "1h"` or `"size": "1kb"`) and for computers
+(eg `"exists_time_in_millis": 3600000` or `"size_in_bytes": 1024`).
+The human-readable values can be turned off by adding `?human=false`
+to the query string. This makes sense when the stats results are
+being consumed by a monitoring tool, rather than intended for human
+consumption.  The default for the `human` flag is
+`false`.
+
+[[hot-threads-api]]
+=== Hot Threads API
+
+experimental[]
+
+//QUESTION: Which of the params shown for the Elasticsearch hot_threads API also available here? (I'm talking about the threads, interval, type, and ignore_idle_threads parameters shown here: https://www.elastic.co/guide/en/elasticsearch/reference/2.2/cluster-nodes-hot-threads.html) I ask even though these parameters are not mentioned in issue #4446 because I tested ?threads, and it works, but when I tested the other params, I couldn't really tell if they were working. 
```

Comment:
The current query string params support by this call are  the following.

`thread` => integer, default 3, limit the number of hot threads returned by the api, they ar ethe threads that consume the most resources.
`human` => boolean default false, return the result in plain text instead of the json format.
`ignore_idle_threads` => default true, will not return idle threads.